src/lint-api.jl

```diff
@@ -0,0 +1,136 @@
+export lint_string
+"""
+    lint_string(str::AbstractString, root = nothing, server = get_symol_server()[1])
+
+Lint the given string and get the result as a CST.
+"""
+function lint_string(str::AbstractString, root = nothing, server = SymbolServer.get_symol_server()[1])
+    # parse
+    cst = CSTParser.parse_code(str)
+
+    # TODO: we should not need to write to a file!
```

Comment:
Yeah, that is definitely key :) LS doesn't write any files, so it can be done, but I'm not familiar enough with the API to help out here.

Reply:
From https://github.com/julia-vscode/StaticLint.jl/blob/master/test/runtests.jl:
```julia
function parse_and_pass(s)
    empty!(server.files)
    f = StaticLint.File("", s, CSTParser.parse(s, true), nothing, server)
    StaticLint.setroot(f, f)
    StaticLint.setfile(server, "", f)
    StaticLint.scopepass(f)
    StaticLint.check_all(f.cst, StaticLint.LintOptions(), server)
    return f.cst
end
```